test/controllers/auth.test.js

```diff
@@ -0,0 +1,81 @@
+const { expect } = require('chai');
+const jwt = require('jsonwebtoken');
+const sinon = require('sinon');
+const mongoose = require('mongoose');
+
+const User = require('../../models/user');
+const AuthController = require('../../controllers/auth');
+
+const next = () => {};
+
+describe('Authentication Controller - Login', () => {
+  it('Should throw an error with status code 500 if database connection failure', async () => {
+    sinon.stub(User, 'findOne');
+    User.findOne.throws();
+
+    const req = {
+      body: {
+        email: 'qa@test.com',
+        password: 'qatest123*'
+      }
+    };
+
+    const result = await AuthController.login(req, {}, next);
+
+    // expect(result).to.be.an('error');
+    // expect(result).to.have.property('statusCode', 500);
+
+    AuthController.login(req, {}, next).then((result) => {
+      expect(result).to.be.an('error');
+      expect(result).to.have.property('statusCode', 500);
+      done();
+    });
+
+    User.findOne.restore();
+  });
+
+  it('Should send a response with a valid user status for an existing user', (done) => {
+    const username = process.env.MONGO_USERNAME;
+    const password = process.env.MONGO_PASSWORD;
+
+    const collectionName = 'test-messages'; //  MongoDB collection name
+    const dbURI = `mongodb+srv://${username}:${password}@cluster0.ypnh4.mongodb.net/${collectionName}`; // MongoDB connection URI
+
+    mongoose
+      .connect(dbURI)
+      .then((result) => {
+        const payload = {
+          email: 'qa@test.com',
+          password: 'qatest123*',
+          name: 'QA',
+          posts: [],
+          _id: 'mock'
```

Comment:
For commit [3053ef9](https://github.com/SuphawitCE/node-restapi/pull/33/commits/3053ef97e6429e0f70fb51de96c92d9067528b88), need to change `_id` to be the same as in MongoDB to test duplicate user